src/sonic-config-engine/minigraph.py

```diff
@@ -2192,12 +2192,16 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
 
         results['DEVICE_METADATA']['localhost']['peer_switch'] = list(results['PEER_SWITCH'].keys())[0]
     elif results['DEVICE_METADATA']['localhost']['type'] == 'SpineRouter':
-        if macsec_enabled == 'True':
-            results['DEVICE_METADATA']['localhost']['subtype'] = 'UpstreamLC'
-        elif macsec_enabled == 'False':
-            results['DEVICE_METADATA']['localhost']['subtype'] = 'DownstreamLC'
+        # If a devive is SpineRouter but not a chassis, then it's a lower spine router
+        if not is_minigraph_for_chassis(chassis_type):
```

Comment:
this may not work. The minigraph of the upperT2 is_minigraph_for_chassis(chassis_type) will also be True.

Reply:
This may not break chassis, because in the minigraph generated in `sonic-mgmt` for T2 `not is_minigraph_for_chassis(chassis_type)` will be True

Reply:
> this may not work. The minigraph of the upperT2 `not is_minigraph_for_chassis(chassis_type) `will also be True.

For UT2, `results['DEVICE_METADATA']['localhost']['type']` will be `UpperSpineRouter`. So we should be good.